src/snowflake/snowpark/modin/plugin/compiler/snowflake_query_compiler.py

```diff
@@ -10491,15 +10492,25 @@ def _make_discrete_difference_expression(
                         self._modin_frame.ordering_column_snowflake_quoted_identifiers
                     )
                 )
-                return (col1 | col2) & (not_(col1 & col2))
+                return SnowparkPandasColumn(
+                    snowpark_column=(col1 | col2) & (not_(col1 & col2)),
```

Comment:
This is interesting! I tracked it here https://snowflakecomputing.atlassian.net/browse/SNOW-1650680